desktop_version/src/FileSystemUtils.cpp

```diff
@@ -707,17 +732,63 @@ void FILESYSTEM_enumerateLevelDirFileNames(
 	}
 }
 
-static void PLATFORM_getOSDirectory(char* output)
+static int PLATFORM_getOSDirectory(char* output, const size_t output_size)
 {
 #ifdef _WIN32
 	/* This block is here for compatibility, do not touch it! */
 	WCHAR utf16_path[MAX_PATH];
-	SHGetFolderPathW(NULL, CSIDL_PERSONAL, NULL, SHGFP_TYPE_CURRENT, utf16_path);
-	WideCharToMultiByte(CP_UTF8, 0, utf16_path, -1, output, MAX_PATH, NULL, NULL);
+	HRESULT retcode = SHGetFolderPathW(
+		NULL,
+		CSIDL_PERSONAL,
+		NULL,
+		SHGFP_TYPE_CURRENT,
+		utf16_path
+	);
+	int num_bytes;
+
+	if (retcode != S_OK)
```

Comment:
This should use `SUCCEEDED()` or `FAILED()`, HRESULT is really weird and needs these macros to be reliable

Reply:
Weird - I didn't find any mention of this totally very excellent API design when I looked up `HRESULT` for the first time. Regardless: Fixed, rebased, force-pushed.